### PR TITLE
Added Open as tab toggle, test cases, and functionality

### PR DESCRIPTION
**Pull request by NyWn**
Haven't written much vuejs before, so I may have misused the store in the `.vue` components:
https://github.com/NyWn/daily-apps/blob/dfa6baba0fc75567132c72afa80c29a694a89c3c/packages/components/src/components/DaCard.vue#L84
https://github.com/NyWn/daily-apps/blob/dfa6baba0fc75567132c72afa80c29a694a89c3c/packages/components/src/components/DaInsanePost.vue#L79

@idoshamun in your message explaining what to do, you mentioned `compatibility/settings.ts`, however I couldn't find this file, unless its `__tests__/DaSettings.js. 

I've probably missed something or broken some convention 😉 , but I reckon this is a decent starting point..

### CODE REVIEW

**Comment by idoshamun** [comment]
Hi @NyWn,

Thanks for dedicating the time for contributing to daily.dev :)

Quickly glancing through I have some initial notes:
* Please make sure to sign the CLA with both your emails/ids, if it's not possible please let me know.
* `compatibility/settings.ts` is in the [daily-api](https://github.com/dailydotdev/daily-api/tree/master/src/compatibility) repo, but I can take care of it.
* Please note that your build failed, let me know if you need assistance.
* I noticed you use the store in `DaCard` and `DaInsanePost`. Unfortunately, it is not possible as these components should be standalone. Please set a new property and provide this parameter from the outside.

I will review more carefully once I have some free time.

File: packages/extension/src/components/DaSettings.vue [hunk 1]
```diff
@@ -75,6 +77,10 @@ export default {
       this.$store.commit('ui/setShowOnlyNotReadPosts', val);
       this.refreshFeed();
     },
+    async toggleOpenInTab(val) {
```
Comment (idoshamun) [inline packages/extension/src/components/DaSettings.vue:80]: This function doesn't have to be `async`. Nothing `async` really going on inside it

File: packages/extension/src/store/modules/ui.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ const initialState = () => ({
   lastBannerSeen: new Date(),
   showPremium: false,
   showNewSource: false,
+  openNewTab: false,
```
Comment (idoshamun) [inline packages/extension/src/store/modules/ui.js:21]: I think it's better to set it to `true` by default.
Many users asked why we changed this behavior.

File: packages/extension/src/store/plugins/sync.js [hunk 3]
```diff
@@ -106,6 +108,10 @@ const plugin = (store) => {
       // TODO: handle error
       await syncSettings(state);
       break;
+    case 'ui/setOpenNewTab':
```
Comment (idoshamun) [inline packages/extension/src/store/plugins/sync.js:111]: You can add this case to the above block of settings.
It's better to concentrate all the settings-based mutations in one place.

**Comment by NyWn** [comment]
> Hi @NyWn,
> 
> Thanks for dedicating the time for contributing to daily.dev :)
> 
> Quickly glancing through I have some initial notes:
> 
> * Please make sure to sign the CLA with both your emails/ids, if it's not possible please let me know.
> * `compatibility/settings.ts` is in the [daily-api](https://github.com/dailydotdev/daily-api/tree/master/src/compatibility) repo, but I can take care of it.
> * Please note that your build failed, let me know if you need assistance.
> * I noticed you use the store in `DaCard` and `DaInsanePost`. Unfortunately, it is not possible as these components should be standalone. Please set a new property and provide this parameter from the outside.
> 
> I will review more carefully once I have some free time.

Hey - Thanks for the feedback!

Regarding the CLA - its not possible to sign with the "Nick Wilkin" one unfortunately.

I'll look through the build failure later this evening.

I had a feeling that store didn't belong in the components like that, but wasn't sure how to do it properly. Will look into passing it as a property

Will also review your recommended changes and correct accordingly this evening.

Thanks!

**Comment by idoshamun** [comment]
I whitelisted "Nick Wilkin" in the CLA system, it should be fine now.

**Comment by sshanzel** [comment]
I can try and work on the API side if you guys don't mind 😁

**Comment by idoshamun** [comment]
This would be great! Go ahead :)
Thanks man!

**Comment by sshanzel** [comment]
Related [PR](https://github.com/dailydotdev/daily-api/pull/118) for the API side.

**Comment by idoshamun** [comment]
Hi @NyWn, do you need help with proceeding with the PR?

**Comment by sshanzel** [comment]
Once I'm finished with the API later I can also offer a helping hand here.

**Comment by NyWn** [comment]
Hey @idoshamun, sorry for radio silence - been running at 200% with work. I'm hoping to get around to this preferably by end of day today / tomorrow. If I don't manage to find the time, the help would be appreciated, thank you @sshanzel.

**Comment by idoshamun** [comment]
Thanks for letting me know. Actually you can revert your changes to `DaCard.vue` and `DaInsanePost.vue`.
I'm rewriting them both due to major changes in design.

**Comment by sshanzel** [comment]
> If I don't manage to find the time, the help would be appreciated, thank you @sshanzel.

Hi @NyWn, sure let's work on this together once the major changes have been applied. Thank you!

**Comment by idoshamun** [comment]
Hi @NyWn and @sshanzel,
I'm done with most of my major changes, my advice is to rollback `DaCard.vue` and `DaInsanePost.vue` and implement again. Please take a look also at the `postMixin.js`, it should be your starting point.

**Comment by sshanzel** [comment]
Hi @NyWn, I think I don't have yet the necessary permissions to make the changes on your forked repo. I'll probably create a separate PR if you don't mind and just follow where you left off.

**Comment by idoshamun** [comment]
@NyWn thanks for working hard on this PR!
@sshanzel followed your path and completed the work 🔥

**Comment by sshanzel** [comment]
@NyWn thanks man! 🍻

**Comment by NyWn** [comment]
Thank you @sshanzel! Once again, apologies @idoshamun for not seeing it through completely, real-life demands really picked up suddenly..! Glad to see the change implemented 😃

**Comment by idoshamun** [comment]
@NyWn don't you worry about it!

**Pull request closed without merging**